src/tagged-type.ts

```diff
@@ -66,7 +66,7 @@ class TaggedTypeEncoded {
       const _out: { [key: string]: any } = {};
       for (const k in input) {
         if (k.startsWith("@")) {
-          _out[k] = input[k];
+          _out["@object"] = k == "@object" ? input[k] : { [k]: input[k] };
```

Comment:
We still need to call `TaggedTypeFormat.encode` on the value if it's a user-defined field.